.circleci/clean_appeng_versions.sh

```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env bash
+
+# this is complicated because gnu date and BSD date have different implementations
+STALEDATE=$(date --version &>/dev/null && date --date="5 days ago" +"%Y-%m-%d" || date -v-5d +"%Y"-"%m"-"%d")
+echo ... will remove all STOPPED AppEngine versions older than $STALEDATE
+
+gcloud --project=$GOOGLE_PROJECT_ID app versions list \
+    --format json \
+    --filter="version.createTime.date('%Y-%m-%d', Z)<$STALEDATE AND version.servingStatus=STOPPED" |\
```

Comment:
WOW, I think what you really want to put here is this `--filter="(version.createTime < '-p5d') AND version.servingStatus=STOPPED"`
I think you could reduce this file in terms of size and complexity using my approximation. all `google sdk` support iso8601 durations wherever you could use a timestamp.

Reply:
just to help you to understand, the `=` is an operator with arity 2 for comparison not for assignment. Therefore other comparison operators are allowed and DateTime google cloud SDK help page says it supports absolute and durations (relative range dates)  wherever it needs a date.

Reply:
oh yes 👍  this was super awkward